bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/MycroftHandler.java

```diff
@@ -0,0 +1,234 @@
+import org.openhab.binding.mycroft.internal.channels.ChannelCommandHandler;
+import org.openhab.binding.mycroft.internal.channels.DuckChannel;
+import org.openhab.binding.mycroft.internal.channels.FullMessageChannel;
+import org.openhab.binding.mycroft.internal.channels.ListenChannel;
+import org.openhab.binding.mycroft.internal.channels.MuteChannel;
+import org.openhab.binding.mycroft.internal.channels.MycroftChannel;
+import org.openhab.binding.mycroft.internal.channels.SpeakChannel;
+import org.openhab.binding.mycroft.internal.channels.UtteranceChannel;
+import org.openhab.binding.mycroft.internal.channels.VolumeChannel;
+import org.openhab.core.io.net.http.WebSocketFactory;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MycroftHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+@NonNullByDefault
+public class MycroftHandler extends BaseThingHandler implements MycroftConnectionListener {
+
+    private final Logger logger = LoggerFactory.getLogger(MycroftHandler.class);
+
+    protected final MycroftConnection connection;
+    private @Nullable ScheduledFuture<?> scheduledFuture;
+    private MycroftConfiguration config = new MycroftConfiguration();
+    private boolean thingDisposing = false;
+    protected Map<ChannelUID, MycroftChannel<?>> mycroftChannels = new HashMap<>();
+
+    /** The reconnect frequency in case of error */
+    private static final int POLL_FREQUENCY_SEC = 30;
+    private int sometimesSendVolumeRequest = 0;
+
+    public MycroftHandler(Thing thing, WebSocketFactory webSocketFactory) {
+        super(thing);
+        String websocketID = thing.getUID().getAsString().replace(':', '-');
+        if (websocketID.length() < 4) {
+            websocketID = "mycroft-" + websocketID;
+        }
+        if (websocketID.length() > 20) {
+            websocketID = websocketID.substring(websocketID.length() - 20);
+        }
+        this.connection = new MycroftConnection(this, webSocketFactory.createWebSocketClient(websocketID));
+    }
+
+    /**
+     * Stops the API request or websocket reconnect timer
+     */
+    private void stopTimer() {
+        ScheduledFuture<?> future = scheduledFuture;
+        if (future != null) {
+            future.cancel(false);
+            scheduledFuture = null;
+        }
+    }
+
+    /**
+     * Starts the websocket connection.
+     * sometimes send a get volume request to fully test the connection / refresh volume.
+     */
+    private void startWebsocket() {
```

Comment:
The method name is misleading. It is not just used for starting/creating the websocket once, but to keep the connection established as it is called every 30 secs.

Reply:
Done